pkg/apis/core/v1alpha1/storagecluster.go

```diff
@@ -348,7 +348,18 @@ type DataProviderSpec struct {
 type MonitoringSpec struct {
 	// EnableMetrics this exposes the storage cluster metrics to external
 	// monitoring solutions like Prometheus.
-	EnableMetrics bool `json:"enableMetrics,omitempty"`
+	EnableMetrics *bool `json:"enableMetrics,omitempty"`
+	// Prometheus contains the details of the Prometheus stack deployed to monitor
+	// metrics from the storage cluster.
+	Prometheus *PrometheusSpec `json:"prometheus,omitempty"`
```

Comment:
The spec is not explicit enough to distinguish between 
- deploying prometheus operator only
- deploying prometheus and creating a service monitor
- creating only service monitor

Reply:
May be it should be something like this
```
type MonitoringSpec struct {
    Prometheus *PrometheusSpec
}

type PrometheusSpec struct {
       EnableMetrics *bool -> enables exporting metrics to prometheus endpoint(which could be deployed by us)
       DeployComponent *bool -> deploy prometheus stack
       RemoteWriteEndpoint string
}

```

Reply:
We cannot remove EnableMetrics as it is a released flag. Marking it for deprecation. If present copying it to the new flag in Prometheus section and then removing it from the StorageCluster object.
Now the spec looks like this -
```
type MonitoringSpec struct {
    EnableMetrics *bool
    Prometheus *PrometheusSpec
}

type PrometheusSpec struct {
       ExportMetrics bool // new flag that gets set if EnableMetrics was set previously
       Enabled bool 
       RemoteWriteEndpoint string
}
```